src/platform-aws.c

```diff
@@ -580,7 +580,11 @@ int platform_init(const char **provider_filter)
 
 	domain_per_thread = ofi_nccl_domain_per_thread();
 	if (domain_per_thread == -1) {
```

Comment:
why not write this as `if (domain_per_thread == -1 && platform_data != NULL)`?

Reply:
If `domain_per_thread == -1`, we need to set it to a valid value (0 or 1), whether or not `platform_data == NULL`.